Dab animation framework + Hard Dab system + Existing Dab animated (achievement included)

Hopekz opened the pull request:
This PR adds 4 things.
------

**1. A dab proc framework.** 
This means that every single item in the game can perform an animated dab.
Usage for admins would be running the DabAnimation() procedure similar to SpinAnimation() 
Similar to spin animations angle of dab gets plotted on a matrix to animate.
Example: mob.DabAnimation(angle = 70 , loop = 10)  would hit a dab at 70 degrees 10 times.

 **2. Used this new proc to animate the regular dab emote.**
regular *dab hits between 30-45 degrees.

**3. Added a new emote called hard dab.**
This emote is called with *dabb and has tier systems dependant on RNG.
This hard dab hits between 55 to 110 degrees.
**There are 4 tier levels for this dab** each with increasing flair in the form of flavor text and new sound effects.
Hard dab brain damage is calculated based on the angle of dab performed.

**An Ultimate Dab can be achieved with a probability of 4%** 
This ultimate dab not only looks awesome (animation+flair text) and sounds great but also restores brain damage.

**4. Added the "King of Dab" achievement for hitting the ultimate dab.**

Honestly just load up this PR if you're curious about it or want to see the animation.

![357450666265804800](https://user-images.githubusercontent.com/24533979/70386912-56afb080-1964-11ea-9d80-e7240b3ab753.png)

Edit: Added moodlet for hitting the ultimate dab

#### Changelog

:cl:  Hopek
rscadd: Created a Dab framework. Every single sprite in the game can now dab with DabAnimation() . Usage similar to SpinAnimation()
rscadd: Animated *dab using this framework
rscadd: Added a hard dab under *dabb . Hard dabs give more flair the harder the dab (with flavortext and sound effects between tiers)
rscadd: Hit the ultimate dab for extra flex and an achievement!
/:cl:

Hopekz committed:
First commit .

Hopekz committed:
King of Dab

Fixed dab framework.
Added sound effects with teirs of dabs.
added achievment.
made framework work with regular dab.

Hopekz committed:
Thank you Travis

Very cool
(no idea how I didn't catch this I must be tired)

Hopekz committed:
Changed ultimate DAB sound effect (provided by meep)

SomeguyManperson reviewed (approved):
works about as expected

Hopekz committed:
Updated comments for future clarity.

from:
if(!stay) // if stay param is false dab doesn't return
to:
if(!stay) // if stay param is true dab doesn't return

Hopekz committed:
Make Radystorm546 happy

I removed the hidden flag!
Our code is now 1 line more pretty because of this.

Hopekz committed:
Hard dab and ultimate dab updated

Ultimate Dab hits harder with an new 1 second sound effect (from 0.5 seconds).
Felt it was underwhelming for having the chance to die 4-5 times trying to perform this dab.
Hard dab speeds adjusted to feel better.
Tested in-game.

Hopekz committed:
<100 angle dab tier has its own sound effect

For when your charecter dabs REALLY hard.

Hopekz committed:
Silicons don't get an easy achievment

Tested

Hopekz committed:
Added moodlet for hitting the ultimate dab

King of Dabs!
